llvm/include/llvm/IR/Operator.h

```diff
@@ -326,6 +326,21 @@ class FPMathOperator : public Operator {
   /// precision.
   float getFPAccuracy() const;
 
+  /// Returns true if `Ty` is a supported floating-point type for phi, select,
+  /// or call FPMathOperators.
+  static bool isSupportedFloatingPointType(Type *Ty) {
+    if (StructType *StructTy = dyn_cast<StructType>(Ty)) {
```

Comment:
```suggestion
    if (auto *StructTy = dyn_cast<StructType>(Ty)) {
```
Use auto with dyn_cast.